### PR TITLE
implement unfollow user

### PR DESCRIPTION
**Pull request by adrianosela**
closes https://github.com/adrianosela/mapp/issues/99

### CODE REVIEW

File: api/handlers/user.js [hunk 1]
```diff
@@ -225,6 +225,43 @@ let followUser = async function(req, res) {
     }
 };
 
+let unfollowUser = async function(req, res) {
+    try {
+        const userId = req.authorization.id;
+
+        const userToUnfollowId = req.body.userToUnfollowId;
+        if (!userToFollowId) {
```
Comment (israeltq) [inline api/handlers/user.js]: Wrong variable.

File: api/handlers/user.js [hunk 2]
```diff
@@ -225,6 +225,43 @@ let followUser = async function(req, res) {
...
+
+        let user = await User.findById(userId);
+        if (!user) {
+            return res.status(404).send("Requesting user not found");
+        }
+
+        user.following.pull(userToUnfollowId);
+        await User.findByIdAndUpdate(user._id, user, {
+            useFindAndModify: false
+        });
```
Comment (israeltq) [inline api/handlers/user.js]: You could just save the user already found in the previous query

Comment (israeltq) [reply api/handlers/user.js]: no need to find it again

Comment (adrianosela) [reply api/handlers/user.js]: good point. we can do that later. just copy pasta'd the other handler. We should change it in both.

**Pull request merged**